packages/aws-cdk-lib/aws-s3/lib/bucket.ts

```diff
@@ -1882,6 +1886,26 @@ export class Bucket extends BucketBase {
     }
   }
 
+  /**
+   * Set the log group on the singleton AutoDeleteObjects provider lambda.
+   * Note: Calling this multiple times will override the previously set log group
+   *
+   * In the case of no buckets with `autoDeleteObjects: true` exist in the stack,
+   * this will throw an exception as no AutoDeleteObjects provider lambda will exist
+   * in the stack.
+   *
+   * @param scope the stack with bucket(s) with `autoDeleteObjects: true`.
+   * @param logGroup the log group to use on the lambda.
+   */
+  public static setAutoDeleteObjectsLogGroup(scope: Construct, logGroup: logs.ILogGroup): void {
```

Comment:
I don't think this is the right place for this function. Given that the lambda is a singleton for the entire stack, such a function should be at the stack level. Having it exist on the Bucket class and taking in a scope that can be set at the construct level is misleading as to the actual scope of this function.

Reply:
You could argue that you've explained that in the docstring, but I think this will still mislead customers.

Reply:
Also, as mentioned last week, I think that a solution here needs to be implemented so that it is consistent across custom resources.